pkg/storage/trb_request_system.go

```diff
@@ -0,0 +1,92 @@
+package storage
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/lib/pq"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/sqlqueries"
+)
+
+// SetTRBRequestSystems links given System IDs to given TRB Request ID
+// This function opts to take a *sqlx.Tx instead of a NamedPreparer because the SQL calls inside this function are heavily intertwined, and we never want to call them outside the scope of a transaction
+func (s *Store) SetTRBRequestSystems(ctx context.Context, tx *sqlx.Tx, trbRequestID uuid.UUID, systemIDs []string) error {
+	if trbRequestID == uuid.Nil {
+		return errors.New("unexpected nil system intake ID when linking system intake to system id")
+	}
+
+	if _, err := tx.NamedExecContext(ctx, sqlqueries.TRBRequestSystemForm.Delete, map[string]interface{}{
+		"system_ids":     pq.StringArray(systemIDs),
+		"trb_request_id": trbRequestID,
+	}); err != nil {
+		appcontext.ZLogger(ctx).Error("Failed to delete system ids linked to system intake", zap.Error(err))
+		return err
+	}
+
+	// no need to run insert if we are not inserting new system ids
+	if len(systemIDs) < 1 {
+		return nil
+	}
+
+	userID := appcontext.Principal(ctx).Account().ID
+
+	setTRBRequestSystemsLinks := make([]models.TRBRequestSystem, len(systemIDs))
+
+	for i, systemID := range systemIDs {
+		systemIDLink := models.NewTRBRequestSystem(userID)
+		systemIDLink.ID = uuid.New()
+		systemIDLink.ModifiedBy = &userID
+		systemIDLink.TRBRequestID = trbRequestID
+		systemIDLink.SystemID = systemID
+
+		setTRBRequestSystemsLinks[i] = systemIDLink
+	}
+
+	if _, err := tx.NamedExecContext(ctx, sqlqueries.TRBRequestSystemForm.Set, setTRBRequestSystemsLinks); err != nil {
+		appcontext.ZLogger(ctx).Error("Failed to insert linked system intake to system ids", zap.Error(err))
```

Comment:
🙈 `system intake` -> `trb request`

Reply:
Addressed in [a75586c](https://github.com/CMSgov/easi-app/pull/2445/commits/a75586c074fc7670c87ff0ea954f5f07cca9f0d3)